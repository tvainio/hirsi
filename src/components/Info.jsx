import React from "react";
import PropTypes from "prop-types";
import { Panel } from "react-bootstrap";

class Info extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }
  render() {
 //   let { timestamp, id, ttl, ...data } = this.props.data;

    return (
      <div>
        <Panel>
          <Panel.Body>
            You missed this and that...
          </Panel.Body>
        </Panel>
      </div>
    );
  }
}
/*
Info.propTypes = {
  data: PropTypes.object.isRequired,
  title: PropTypes.string.isRequired,
  timestamp: PropTypes.string,
  unit: PropTypes.string
};
*/
export default Info;
