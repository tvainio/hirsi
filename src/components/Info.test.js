import React from "react";
import { shallow } from "enzyme";
import Info from "./Info";
import renderer from "react-test-renderer";

it("shows game over message only if health equals 0", () => {
  expect(
    shallow(<Info health={0} usedLetters={[]} word={""} />)
      .find("#message")
      .text()
  ).toContain("YOU MISSED:");
});
it("shows missed letters", () => {
  expect(
    shallow(<Info health={0} usedLetters={["A", "B"]} word={"TABLET"} />)
      .find("#missed")
      .text()
  ).toBe("T, L, E");
});
