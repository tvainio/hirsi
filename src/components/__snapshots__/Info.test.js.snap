// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    className="panel panel-default"
  >
    <div
      className="panel-body"
    >
      You missed this and that...
    </div>
  </div>
</div>
`;
